services/app-api/src/resolvers/contract/indexContracts.ts

```diff
@@ -0,0 +1,145 @@
+import type { Span } from '@opentelemetry/api'
+import { ForbiddenError } from 'apollo-server-lambda'
+import {
+    isStateUser,
+    isCMSUser,
+    isAdminUser,
+    isBusinessOwnerUser,
+} from '../../domain-models'
+import type { ContractType } from '../../domain-models'
+import { isHelpdeskUser } from '../../domain-models/user'
+import type { QueryResolvers } from '../../gen/gqlServer'
+import { logError, logSuccess } from '../../logger'
+import type { Store } from '../../postgres'
+import { NotFoundError } from '../../postgres'
+import {
+    setErrorAttributesOnActiveSpan,
+    setResolverDetailsOnActiveSpan,
+    setSuccessAttributesOnActiveSpan,
+} from '../attributeHelper'
+import { GraphQLError } from 'graphql/index'
+import type { ContractOrErrorArrayType } from '../../postgres/contractAndRates/findAllContractsWithHistoryByState'
+
+const parseContracts = (
+    contractsWithHistory: ContractOrErrorArrayType,
+    span?: Span
+): ContractType[] => {
+    // separate valid contracts and errors
+    const parsedContracts: ContractType[] = []
+    const errorParseContracts: string[] = []
+    contractsWithHistory.forEach((parsed) => {
+        if (parsed.contract instanceof Error) {
+            errorParseContracts.push(
+                `${parsed.contractID}: ${parsed.contract.message}`
+            )
+        } else {
+            parsedContracts.push(parsed.contract)
+        }
+    })
+
+    // log all contracts that failed parsing to otel.
+    if (errorParseContracts.length > 0) {
+        const errMessage = `Failed to parse the following contracts:\n${errorParseContracts.join(
+            '\n'
+        )}`
+        logError('indexHealthPlanPackagesResolver', errMessage)
+        setErrorAttributesOnActiveSpan(errMessage, span)
+    }
+
+    return parsedContracts
+}
+
+const validateAndReturnContracts = (results: ContractType[], span?: Span) => {
```

Comment:
this function seems misnamed now. I see it came from indexHPP but I don't like that we're calling logSuccess inside of here. Maybe trim it down into something that just returns the node structure? logSuccess in the handler itself?